src/it/scala/io/iohk/ethereum/ledger/BlockImporterItSpec.scala

```diff
@@ -240,4 +240,39 @@ class BlockImporterItSpec
       Thread.sleep(1000); blockchain.getLatestCheckpointBlockNumber() shouldEqual newBlock5.header.number + 1
     }
   }
+
+  it should "ask BlockFetcher to resolve missing node" in {
+    val parent = blockchain.getBestBlock().get
+    val newBlock: Block = getBlock(genesisBlock.number + 5, difficulty = 104, parent = parent.header.hash)
+    val invalidBlock = newBlock.copy(header = newBlock.header.copy(beneficiary = Address(111).bytes))
+
+    val ledger = new TestLedgerImpl(successValidators)
+    val blockImporter = system.actorOf(
+      BlockImporter.props(
+        fetcherProbe.ref,
+        ledger,
+        blockchain,
+        syncConfig,
+        ommersPoolProbe.ref,
+        broadcasterProbe.ref,
+        pendingTransactionsManagerProbe.ref,
+        supervisor.ref
+      )
+    )
+
+    blockImporter ! BlockImporter.Start
+    blockImporter ! BlockFetcher.PickedBlocks(NonEmptyList.fromListUnsafe(List(invalidBlock)))
+
+    eventually {
+      Thread.sleep(1000)
```

Comment:
Why the `Thread.sleep` if you are using `eventually`?

Reply:
I saw the `Thread.sleep` in some of the existing tests, so I reused it here. I will try to remove them and define a custom akka `Patience` for the `eventually`

Reply:
I have tried to use `eventually` without any `Thread.sleep`s and defining implicit/explicit `Patience` but for some reason it does not re-run the test if it fails. This is probably the reason that `Thread.sleep`s are introduced.